qfdmo/management/commands/import_product_from_qfdmod.py

```diff
@@ -0,0 +1,207 @@
+# A partir du fichier : https://docs.google.com/spreadsheets/d/1WIKCP3mye_ZYWVAyKecwHN5xrwaU0EUp/edit?usp=sharing&ouid=114461779294398717478&rtpof=true&sd=true
+
+# si il existe déjà un objet avec le nom, on met à jour l'identifiant_qfdmod
+# on crée les synomimes si ils n'existent pas
+
+# 1 match des sous-categories de la colonne P
+# si pas de colonne P, on crée catégorie colonne B et sous-catégorie de la colonne M
+
+
+import csv
+from functools import reduce
+from operator import or_
+from typing import List
+
+import requests
+from django.core.management.base import BaseCommand
+from django.db.models import Q
+
+from qfdmo.models.categorie_objet import Objet
+
+produits_qfdmod_url = (
+    "https://data.ademe.fr/data-fair/api/v1/datasets/que-faire-de-mes-dechets-produits"
+    "/lines?size=10000&page=1&format=csv"
+)
+
+
+def _get_qfdmod_products():
+    qfdmod_products = []
+
+    # Télécharger le fichier CSV depuis l'URL
+    response = requests.get(produits_qfdmod_url)
+    response.raise_for_status()  # Vérifie si la requête a réussi
+
+    # Lire le fichier CSV depuis le contenu téléchargé
+    content = response.content.decode("utf-8").splitlines()
+    reader = csv.DictReader(content)
+
+    for row in reader:
+        qfdmod_products.append(row)
+
+    return qfdmod_products
+
+
+def _get_product_names_from_qfdmod_product(qfdmod_product: dict) -> List[str]:
+    product_names = [
+        word.lower().strip()
+        for word in qfdmod_product["Synonymes_existants"].split("/")
+        if word.lower().strip()
+    ]
+    product_names.append(qfdmod_product["Nom"].lower().strip())
+    product_names = list(set(product_names))
+    return product_names
+
+
+def _choose_sous_categorie(sous_categories, product_name):
+    choosen_sous_categorie = None
+    while choosen_sous_categorie is None:
+        input_text = f"\nPour le produit : {product_name}"
+        input_text += "\nChoisissez la categorie adéquat ?\n\n"
+        input_text += "\n".join(
+            [
+                f"{i + 1} : {prop.libelle} ({prop.categorie.libelle})"
+                for i, prop in enumerate(sous_categories)
+            ]
+        )
+        input_text += "\n0 : Ignorer\n\n"
+        input_text += "\n\nVotre choix : "
```

Comment:
On pourrait un peu simplifier l'écriture, même si je suis pas 100% sûr que ça simplifie 

```suggestion
        input_text = (
          f"\nPour le produit : {product_name}"
          "\nChoisissez la categorie adéquat ?\n\n"
        ) + "\n".join(
            [
                f"{i + 1} : {prop.libelle} ({prop.categorie.libelle})"
                for i, prop in enumerate(sous_categories)
            ]
        ) + (
        "\n0 : Ignorer\n\n"
        "\n\nVotre choix : "
        )
```

Reply:
Ou alors 
```suggestion
        input_text = "\n".join([
          f"Pour le produit : {product_name}",
          "Choisissez la categorie adéquat ?\n\n",
          *[
                f"{i + 1} : {prop.libelle} ({prop.categorie.libelle})"
                for i, prop in enumerate(sous_categories)
          ],
          "0 : Ignorer\n\n",
          "\nVotre choix : ",
        ])
```